src/Service.Tests/Configuration/HotReload/AuthorizationResolverHotReloadTests.cs

```diff
@@ -0,0 +1,215 @@
+            Runtime: new(
+                Rest: new(Enabled: true),
+                GraphQL: new(), // GraphQL doesn't yet support hot-reload
+                Host: hostOptions
+            ),
+            Entities: new(entityMap));
+
+        File.WriteAllText(
+            path: fileName,
+            contents: runtimeConfig.ToJson());
+    }
+
+    /// <summary>
+    /// Create initial configuration file and start the test server.
+    /// The testserver is used in all tests to validate the hot-reload behavior
+    /// as each test will overwrite this configuration file.
+    /// </summary>
+    [ClassInitialize]
+    public static async Task ClassInitializeAsync(TestContext context)
+    {
+        // Arrange
+        EntityActionFields initialFieldPermissions = new(
+            Include: new HashSet<string>()
+            {
+                "id",
+                "publisher_id"
+            },
+            Exclude: new HashSet<string>()
+            {
+                "title"
+            }
+        );
+
+        EntityAction actionForRole = new(
+            Action: EntityActionOperation.Read,
+            Fields: initialFieldPermissions,
+            Policy: null
+            );
+
+        EntityPermission permissions = new(
+            Role: "Role1",
+            Actions: new[] { actionForRole }
+        );
+
+        // At least one entity is required in the runtime config for the engine to start.
+        // Even though this entity is not under test, it must be supplied to the config
+        // file creation function.
+        Entity requiredEntity = new(
+            Source: new("books", EntitySourceType.Table, null, null),
+            Rest: new(Enabled: true),
+            GraphQL: new(Singular: "", Plural: "", Enabled: false),
+            Permissions: new[] { permissions },
+            Relationships: null,
+            Mappings: null);
+
+        Dictionary<string, Entity> entityMap = new()
+        {
+            { INITIAL_ENTITY_NAME, requiredEntity }
+        };
+
+        CreateCustomConfigFile(fileName: AUTHZ_HR_FILENAME, entityMap);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={AUTHZ_HR_FILENAME}"
+        };
+
+        _testServer = new(Program.CreateWebHostBuilder(args));
+        _testClient = _testServer.CreateClient();
+
+        // Setup validation - ensure initial config is honored.
+        HttpRequestMessage initialRequest = new(HttpMethod.Get, $"{RestRuntimeOptions.DEFAULT_PATH}/{INITIAL_ENTITY_NAME}?$select=id,publisher_id");
+        initialRequest.Headers.Add("X-MS-API-ROLE", "Role1");
+        HttpResponseMessage initialResponse = await _testClient.SendAsync(initialRequest);
+
+        Assert.AreEqual(
+            expected: HttpStatusCode.OK,
+            actual: initialResponse.StatusCode,
+            message: "Initial configuration bootstrap failed.");
+    }
+
+    [ClassCleanup]
+    public static void ClassCleanup()
+    {
+        _testServer.Dispose();
```

Comment:
Can delete the config file that was only created for these tests in cleanup.

Reply:
updated